pkg/endpoints/pipeline.go

```diff
@@ -59,6 +67,25 @@ type BuildRequest struct {
 	BRANCH   string `json:"branch"`
 }
 
+// ResourceBinding - a name and a reference to a resource
+type ResourceBinding struct {
+	BINDINGNAME     string `json:"bindingname"`
+	RESOURCEREFNAME string `json:"resourcerefname"`
+}
+
+//ManualPipelineRun - represents the data required to create a new PipelineRun
+type ManualPipelineRun struct {
+	PIPELINENAME      string `json:"pipelinename"`
+	PIPELINERUNTYPE   string `json:"pipelineruntype,omitempty"`
+	GITRESOURCENAME   string `json:"gitresourcename,omitempty"`
+	IMAGERESOURCENAME string `json:"imageresourcename,omitempty"`
+	GITCOMMIT         string `json:"gitcommit,omitempty"`
+	REPONAME          string `json:"reponame,omitempty"`
+	REPOURL           string `json:"repourl,omitempty"`
+	HELMSECRET        string `json:"helmsecret,omitempty"`
+	REGISTRYSECRET    string `json:"registrysecret,omitempty"`
+}
```

Comment:
I have the same comment regarding the `DOCKER_REGISTRY_LOCATION` environment variable that I had for the `PIPELINE_RUN_SERVICE_ACCOUNT` (https://github.com/tektoncd/dashboard/pull/33/files#r274901277). It seems unnecessary to restrict the user to one docker registry, when we can add this as a field in the `ManualPipelineRun` struct and allow users to use any docker registry they please.

Again, let me know if I'm missing out on a bigger-picture here.

Reply:
I think you're right here too, with this new implementation it makes sense to allow the user to provide a value for this rather than having to hardcode one at install time!